### ISSUE TITLE
Ignore_list regular expression not working

### ISSUE THREAD

**Issue by scheuk** [puppetlabs/beaker#566]
This one took me a while to track down, but my module is located in a directory like this:
/users/Username/p4/puppet/modules/modulename
And beaker was not coping my module to the host/nodes.
I was receiving this error when trace logging enabled:
"After rejecting ignored files/dirs, going to scp []"

Digging into the code, on line 314 of lib/beaker/host.rb:
```
      if has_ignore
        ignore_arr = Array(options[:ignore]).map do |entry|
          "((\/|\\A)#{entry}(\/|\\z))".sub(/\./, "\.") # <- .sub is not substituting "." with "\."
        end
        ignore_re = Regexp.new(ignore_arr.join('|'))
      end
```

irb output:

`
2.1.5 :001 > "((\/|\\A)..(\/|\\z))".sub(/\./, "\.")
 => "((/|\\A)..(/|\\z))"
`

This is causing the ignore directory matcher to match /p4/ (a 2 character directory name), with all of the directories and file paths with /p4/ in them, it was ignoring all of them.

line 337 of lib/beaker/host.rb:
```
        dir_source = Dir.glob("#{source}/**/*").reject do |f|
          f =~ ignore_re
        end
        @logger.trace "After rejecting ignored files/dirs, going to scp [#{dir_source.join(", ")}]"
```

unfortunately I don't have the knowledge to fix the problem,
but I played around with different .sub() parameters and was unable to produce what's needed.

**Comment by chris-reeves** [comment]
This has been fixed in: #578

**Comment by anodelman** [comment]
Merged.

Released in beaker 2.2
